api/envoy/data/accesslog/v3/accesslog.proto

```diff
@@ -201,6 +201,9 @@ message ResponseFlags {
 
       // The request was denied by the external authorization service.
       EXTERNAL_SERVICE = 1;
+
+      // The request was denied by the RBAC filter.
```

Comment:
Would a mechanism to signal which filter rejected in general make more sense? @envoyproxy/api-shepherds

Reply:
@envoyproxy/api-shepherds any suggestions? I think it requires a bit more design in the API for a more general mechanism because the filter name is too long to put in the response flag. Should we open an new issue for this?

Reply:
Why does it need to be a response flag? Maybe just add a string to the parent message giving the filter name that is responsible?

Reply:
oh, I'm just not sure where to output the filter name so just guessed it based on the change in this PR. Do you mean to add a new operator in access log (https://www.envoyproxy.io/docs/envoy/latest/configuration/observability/access_log/usage.html?highlight=access%20log#command-operators), something like `%RESPONSE_REJECT_FILTER%`?

Reply:
Yeah, plausibly this would be the general solution which would make sense across filters. It's even possible that we could have a `sendLocalReply` that is aware of filter context and autopopulates this, but I don't think we need that in the initial version.

Reply:
I created a feature request for this (https://github.com/envoyproxy/envoy/issues/13100), do we have other comments of the API change of this PR? thanks.

Reply:
Can we just do #13100 in this PR? I don't see why we should add in special case flags for one specific filter here. CC @envoyproxy/api-shepherds

Reply:
Can we do both? 

My understanding is that we have a dedicated message [Unauthorized reason](https://github.com/envoyproxy/envoy/blob/8fc484325df6b0c95d9c0e2cb6161d25851574f9/api/envoy/data/accesslog/v3/accesslog.proto#L194-L207) for access control related deny in Envoy, currently it includes the `EXTERNAL_SERVICE` for ext_authz filter and RBAC is for the RBAC filter.

It also benefits as we can reuse the very common `%RESPONSE_FLAGS%` operator in the access log instead of adding a new operator (e.g. `%RESPONSE_REJECT_FILTER%` and we need to carefully make sure it is only used in newer Envoy) to get this information as we're  more interested in unauthorized deny compared to normal filter deny.

Reply:
I think `EXTERNAL_SERVICE` is sufficiently general, RBAC is pretty specific. If you want to put in a general `INTERNAL` that's probably fine, but I'm really thinking we should do this the right way and the cost of doing so is very low.

Reply:
Hmm, my understanding is that the `EXTERNAL_SERVICE` is actually dedicated for the ext_authz filter, its comment actually says `denied by the external authorization service.`. The name looks general but I doubt if it's going to be used by other filters. `INTERNAL` means it can be set by other filters so we cannot rely on it to say a request is rejected by RBAC. 

I'm mainly following the suggestions in https://github.com/envoyproxy/envoy/issues/12873, but @mandarjog can you comment if telemetry is good with this if we add  a new access log operator (e.g. `%RESPONSE_REJECT_FILTER%`) instead of a new response flag (`RBAC`)? I guess it should be good but with extra work to use the new log operator, pass the access log to the telemetry backend and parse it to extract the filter names of RBAC.

Reply:
I think it makes sense to record that the request `served` locally. The response code is already recorded. How about `RESPONDED_BY_FILTER`  instead of `RESPONSE_REJECT_FILTER` ? we could respond with `200` for example.
 
This and `%RESPONSE_CODE_DETAIL%` would give us adequate information.

Reply:
In fact we can retire response flag `response_from_cache_filter` and use `response_from_filter` with `RESPONDED_BY_FILTER_NAME`.

Reply:
@htuch I reverted the API changes to make this PR smaller and easier to merge. Do you have any suggestions for @mandarjog's suggestion, it's a new general response flag for if a filter send some response for a request, could also deprecate the `RESPONSE_CODE_DETAIL`, we could do this in a separate PR as it's no longer RBAC related. also cc @alyssawilk